httpserver/http_connection.h

```diff
@@ -50,27 +50,32 @@ class HttpClientSession
     using tcp = boost::asio::ip::tcp;
 
     boost::asio::io_context& ioc_;
+    tcp::socket socket_;
+    tcp::endpoint endpoint_;
+    http_callback_t callback_;
+    boost::asio::steady_timer deadline_timer_;
+
     boost::beast::flat_buffer buffer_;
     request_t req_;
     response_t res_;
 
-    http_callback_t callback_;
-
     bool used_callback_ = false;
 
+    void on_connect();
+
     void on_write(boost::system::error_code ec, std::size_t bytes_transferred);
 
     void on_read(boost::system::error_code ec, std::size_t bytes_transferred);
 
     void init_callback(std::shared_ptr<std::string>&& body);
 
   public:
-    tcp::socket socket_;
     // Resolver and socket require an io_context
-    explicit HttpClientSession(boost::asio::io_context& ioc,
-                               const request_t& req, http_callback_t cb);
+    HttpClientSession(boost::asio::io_context& ioc, const tcp::endpoint& ep,
```

Comment:
Just curious why `explicit` was removed?

Reply:
It is only necessary for constructors that take 1 argument, so you couldn't do `HttpClientSession = ioc;`. Now that we have many arguments it is not needed.

Reply:
It still prevents doing `HttpClientSession session = {ioc, endpoint, ...}` tho right?

Reply:
I don't think you can do that with or without the `explicit` keyword.

Reply:
Ok, let's keep the removal of explicit :)

Reply:
Just to summarise, it turns out it did prevent `HttpClientSession session = {ioc, endpoint, ...}`, but that shouldn't be a problem anyway.